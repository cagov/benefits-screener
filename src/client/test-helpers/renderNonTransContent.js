import React from "react";
import { shallow } from "enzyme";
import { I18nextProvider, Trans } from "react-i18next";
import i18n from "./i18nForTests";
import shallowUntilTarget from "./shallowUntilTarget";

function renderNonTransContent(Component, props = {}) {
  const providersWrapper = shallow(
    <I18nextProvider i18n={i18n}>
      <Component {...props} />
    </I18nextProvider>
  );

  const wrapper = shallowUntilTarget(providersWrapper, Component.name);
  // Incorrectly using renderNonTransContent to render a component that
  // contains a Trans component results in a silent failure (the rendered
  // snapshot will contain the Trans component itself instead of the rendered
  // contents). This test makes that silent failure more obvious.
  expect(wrapper.containsMatchingElement(<Trans />)).toBe(false);

  return wrapper;
}

module.exports = renderNonTransContent;
