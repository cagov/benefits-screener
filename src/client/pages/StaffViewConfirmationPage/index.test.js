import renderNonTransContent from "../../test-helpers/renderNonTransContent";
import Component from "./index";
import AUTH_STRINGS from "../../../data/authStrings";

describe("<StaffViewConfirmationPage />", () => {
  it("when claimant not started", async () => {
    const wrapper = renderNonTransContent(
      Component,
      "StaffViewConfirmationPage",
      {
        userData: {
          hasLoggedIn: false,
          status: AUTH_STRINGS.statusCode.ok,
          weeksToCertify: [0, 1, 2, 5, 6],
          confirmationNumber: "CONFIRMATION_NUMBER",
          programPlan: ["PUA full time"],
        },
        setUserData: () => {},
      }
    );

    expect(wrapper).toMatchSnapshot();
  });

  it("when claimant in progress", async () => {
    const wrapper = renderNonTransContent(
      Component,
      "StaffViewConfirmationPage",
      {
        userData: {
          hasLoggedIn: true,
          status: AUTH_STRINGS.statusCode.ok,
          weeksToCertify: [0, 1, 2, 5, 6],
          programPlan: ["UI full time"],
        },
        setUserData: () => {},
      }
    );

    expect(wrapper).toMatchSnapshot();
  });

  it("when claimant completed", async () => {
    const wrapper = renderNonTransContent(
      Component,
      "StaffViewConfirmationPage",
      {
        userData: {
          confirmationNumber: "33a6f278-eaa3-41c5-8d08-276fa0364833",
          status: AUTH_STRINGS.statusCode.ok,
          weeksToCertify: [0, 1, 2, 5, 6],
          programPlan: ["UI full time"],
        },
        setUserData: () => {},
      }
    );

    expect(wrapper).toMatchSnapshot();
  });
});
