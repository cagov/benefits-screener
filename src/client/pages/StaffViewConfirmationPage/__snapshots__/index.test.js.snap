// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<StaffViewConfirmationPage /> when claimant completed 1`] = `
<div
  id="overflow-wrapper"
>
  <Header />
  <main
    className="pb-5"
    id="certification-page"
  >
    <div
      className="container p-4"
    >
      <h1>
        Retroactive Certification for Unemployment
      </h1>
      <h3>
        Staff View
      </h3>
      <Button
        active={false}
        as={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "displayName": "Link",
            "propTypes": Object {
              "innerRef": [Function],
              "onClick": [Function],
              "replace": [Function],
              "target": [Function],
              "to": [Function],
            },
            "render": [Function],
          }
        }
        className="text-dark bg-light"
        disabled={false}
        to="/retroactive-certification"
        type="button"
        variant="outline-secondary"
      >
        Search New Claimant
      </Button>
      <h3
        className="mt-5"
      >
        Certification Status
      </h3>
      <CertificationStatus />
      <Button
        active={false}
        className="text-dark bg-light"
        disabled={false}
        onClick={[Function]}
        type="button"
        variant="outline-secondary"
      >
        Print
      </Button>
      <h3
        className="mt-5"
      >
        Certification Weeks
      </h3>
      <ListOfCertifications
        userData={
          Object {
            "confirmationNumber": "33a6f278-eaa3-41c5-8d08-276fa0364833",
            "programPlan": Array [
              "UI full time",
            ],
            "status": "ok",
            "weeksToCertify": Array [
              0,
              1,
              2,
              5,
              6,
            ],
          }
        }
      />
      <h3
        className="mt-5"
      >
        Retroactive Certification Application Details
      </h3>
      <p>
        The claimant is shown the following on the Retroactive Certification Application.
      </p>
      <ClaimantContent />
    </div>
  </main>
  <Footer
    backToTopTag="certification-page"
  />
</div>
`;

exports[`<StaffViewConfirmationPage /> when claimant in progress 1`] = `
<div
  id="overflow-wrapper"
>
  <Header />
  <main
    className="pb-5"
    id="certification-page"
  >
    <div
      className="container p-4"
    >
      <h1>
        Retroactive Certification for Unemployment
      </h1>
      <h3>
        Staff View
      </h3>
      <Button
        active={false}
        as={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "displayName": "Link",
            "propTypes": Object {
              "innerRef": [Function],
              "onClick": [Function],
              "replace": [Function],
              "target": [Function],
              "to": [Function],
            },
            "render": [Function],
          }
        }
        className="text-dark bg-light"
        disabled={false}
        to="/retroactive-certification"
        type="button"
        variant="outline-secondary"
      >
        Search New Claimant
      </Button>
      <h3
        className="mt-5"
      >
        Certification Status
      </h3>
      <CertificationStatus />
      <Button
        active={false}
        className="text-dark bg-light"
        disabled={false}
        onClick={[Function]}
        type="button"
        variant="outline-secondary"
      >
        Print
      </Button>
      <h3
        className="mt-5"
      >
        Certification Weeks
      </h3>
      <ListOfCertifications
        userData={
          Object {
            "hasLoggedIn": true,
            "programPlan": Array [
              "UI full time",
            ],
            "status": "ok",
            "weeksToCertify": Array [
              0,
              1,
              2,
              5,
              6,
            ],
          }
        }
      />
      <h3
        className="mt-5"
      >
        Retroactive Certification Application Details
      </h3>
      <p>
        The claimant is shown the following on the Retroactive Certification Application.
      </p>
      <ClaimantContent />
    </div>
  </main>
  <Footer
    backToTopTag="certification-page"
  />
</div>
`;

exports[`<StaffViewConfirmationPage /> when claimant not started 1`] = `
<div
  id="overflow-wrapper"
>
  <Header />
  <main
    className="pb-5"
    id="certification-page"
  >
    <div
      className="container p-4"
    >
      <h1>
        Retroactive Certification for Unemployment
      </h1>
      <h3>
        Staff View
      </h3>
      <Button
        active={false}
        as={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "displayName": "Link",
            "propTypes": Object {
              "innerRef": [Function],
              "onClick": [Function],
              "replace": [Function],
              "target": [Function],
              "to": [Function],
            },
            "render": [Function],
          }
        }
        className="text-dark bg-light"
        disabled={false}
        to="/retroactive-certification"
        type="button"
        variant="outline-secondary"
      >
        Search New Claimant
      </Button>
      <h3
        className="mt-5"
      >
        Certification Status
      </h3>
      <CertificationStatus />
      <Button
        active={false}
        className="text-dark bg-light"
        disabled={false}
        onClick={[Function]}
        type="button"
        variant="outline-secondary"
      >
        Print
      </Button>
      <h3
        className="mt-5"
      >
        Certification Weeks
      </h3>
      <ListOfCertifications
        userData={
          Object {
            "confirmationNumber": "CONFIRMATION_NUMBER",
            "hasLoggedIn": false,
            "programPlan": Array [
              "PUA full time",
            ],
            "status": "ok",
            "weeksToCertify": Array [
              0,
              1,
              2,
              5,
              6,
            ],
          }
        }
      />
      <h3
        className="mt-5"
      >
        Retroactive Certification Application Details
      </h3>
      <p>
        The claimant is shown the following on the Retroactive Certification Application.
      </p>
      <ClaimantContent />
    </div>
  </main>
  <Footer
    backToTopTag="certification-page"
  />
</div>
`;
